lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,170 @@ def __exit__(self, exc_type, exc_value, traceback):
+                                 True)
+    _funcs['quadratic'] = _FuncInfo(lambda x: x**2,
+                                    lambda x: x**(1. / 2),
+                                    True)
+    _funcs['cubic'] = _FuncInfo(lambda x: x**3,
+                                lambda x: x**(1. / 3),
+                                True)
+    _funcs['sqrt'] = _FuncInfo(lambda x: x**(1. / 2),
+                               lambda x: x**2,
+                               True)
+    _funcs['cbrt'] = _FuncInfo(lambda x: x**(1. / 3),
+                               lambda x: x**3,
+                               True)
+    _funcs['log10'] = _FuncInfo(lambda x: np.log10(x),
+                                lambda x: (10**(x)),
+                                False)
+    _funcs['log'] = _FuncInfo(lambda x: np.log(x),
+                              lambda x: (np.exp(x)),
+                              False)
+    _funcs['x**{p}'] = _FuncInfo(lambda x, p: x**p[0],
+                                 lambda x, p: x**(1. / p[0]),
+                                 True)
+    _funcs['root{p}(x)'] = _FuncInfo(lambda x, p: x**(1. / p[0]),
+                                     lambda x, p: x**p,
+                                     True)
+    _funcs['log10(x+{p})'] = _FuncInfo(lambda x, p: np.log10(x + p[0]),
+                                       lambda x, p: 10**x - p[0],
+                                       True)
+    _funcs['log(x+{p})'] = _FuncInfo(lambda x, p: np.log(x + p[0]),
+                                     lambda x, p: np.exp(x) - p[0],
+                                     True)
+    _funcs['log{p}(x+{p})'] = _FuncInfo(lambda x, p: (np.log(x + p[1]) /
+                                                      np.log(p[0])),
+                                        lambda x, p: p[0]**(x) - p[1],
+                                        True)
+
+    def __init__(self, str_func):
+        """
+        Parameters
+        ----------
+        str_func : string
+            String to be parsed.
+
+        """
+        try:  # For python 2.7 and python 3+ compatibility
+            is_str = isinstance(str_func, basestring)
+        except NameError:
+            is_str = isinstance(str_func, str)
+
+        if not is_str:
+            raise ValueError("The argument passed is not a string.")
+        self._str_func = str_func
+        self._key, self._params = self._get_key_params()
+        self._func = self.get_func()
+
+    def get_func(self):
+        """
+        Returns the _FuncInfo object, replacing the relevant parameters if
+        necessary in the lambda functions.
+
+        """
+
+        func = self._funcs[self._key].copy()
+        if len(self._params) > 0:
+            m = func.direct
+            func.direct = (lambda x, m=m: m(x, self._params))
+            m = func.inverse
+            func.inverse = (lambda x, m=m: m(x, self._params))
+        return func
+
+    def get_directfunc(self):
```

Comment:
these should probably be [properties](http://www.programiz.com/python-programming/property) instead of getters